qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -190,26 +190,25 @@ def get_qubit_operators(
         """
 
         main_operator, aux_second_q_operators = problem.second_q_ops()
-
-        # 1. Convert the main operator (hamiltonian) to QubitOperator and apply two qubit reduction
         num_particles = getattr(problem, "num_particles", None)
 
+        # 1. Convert the main operator (hamiltonian) to a Qubit Operator and apply two qubit reduction
+
         if isinstance(self.qubit_converter, QubitConverter):
-            main_op = self.qubit_converter.convert_only(
-                main_operator,
-                num_particles=num_particles,
-            )
+            self.qubit_converter.force_match(num_particles=num_particles)
+            main_op = self.qubit_converter.convert_only(main_operator, num_particles=num_particles)
         else:
+            if isinstance(self.qubit_converter, ParityMapper):
+                self.qubit_converter.num_particles = num_particles
```

Comment:
Why is overriding the `num_particles` necessary here? Is this just a precaution against potentially incomplete/inconsistent setup of the mapper by the user?

Reply:
This is not really needed as future Mappers will be instantiated from the problem and thus won't need this. However at the moment qeom forces the num_particles of its Problem to the mapper and I tried to follow this logic. This is relevant if the Mapper is set with a wrong particle number for example. 
I'm not sure we want to keep it

Reply:
Okay I see. I suppose we can keep it for now and then see in the future how this plays out :+1: